outputs.tf

```diff
@@ -14,52 +14,62 @@
  * limitations under the License.
  */
 
+output "service_account" {
+  description = "Service account resource (for single use)."
+  value       = google_service_account.service_accounts[0]
+}
+
 output "email" {
-  description = "Service account email (single-use case)."
+  description = "Service account email (for single use)."
   value       = google_service_account.service_accounts[0].email
 }
 
 output "iam_email" {
-  description = "IAM-format service account email (single-use case)."
+  description = "IAM-format service account email (for single use)."
   value       = "serviceAccount:${google_service_account.service_accounts[0].email}"
 }
 
+output "key" {
+  description = "Service account key (for single use)."
+  value       = data.template_file.keys[0].rendered
+}
+
+output "service_accounts" {
+  description = "Service account resources."
+  value       = google_service_account.service_accounts
+}
+
 output "emails" {
-  description = "Map of service account emails."
-  value       = zipmap(var.names, google_service_account.service_accounts.*.email)
+  description = "Service account emails."
+  value       = zipmap(var.names, google_service_account.service_accounts[*].email)
 }
 
 output "iam_emails" {
   description = "IAM-format service account emails."
-  value = zipmap(
-    var.names,
-    formatlist(
-      "serviceAccount:%s",
-      google_service_account.service_accounts.*.email,
-    ),
-  )
+  value       = zipmap(var.names, local.iam_emails)
+}
+
+output "emails_list" {
+  description = "Service account emails."
+  value       = google_service_account.service_accounts[*].email
+}
+
+output "iam_emails_list" {
+  description = "IAM-format service account emails."
+  value       = [for s in google_service_account.service_accounts : "serviceAccount:${s.email}"]
 }
 
 data "template_file" "keys" {
   count    = length(var.names)
   template = "$${key}"
 
   vars = {
-    key = var.generate_keys ? base64decode(
-      element(
-        concat(google_service_account_key.keys.*.private_key, [""]),
-        count.index,
-      ),
-    ) : ""
+    key = var.generate_keys ? base64decode(google_service_account_key.keys[count.index].private_key) : ""
   }
 }
 
 output "keys" {
   description = "Map of service account keys."
   sensitive   = true
-  value = zipmap(
-    formatlist("%s-key.json", var.names),
-    data.template_file.keys.*.rendered,
-  )
+  value       = zipmap(var.names, data.template_file.keys[*].rendered)
```

Comment:
This looks like a breaking change. We're changing the keys from `{names}-key.json` to `names` directly. Do we think this change is worth incrementing by a major version?

Reply:
I don't think so, as those outputs are a) security sensitive so I don't think many people use them, and b) absolutely not meant to be used as input in other resources. The only thing that could break are scripts that rely on those to extract the keys.

TL;DR: I can revert, or we can bump to a major version if you think the change warrants it. I kind of don't but am fine with either.

Reply:
I think we can go ahead and increment by a major version to avoid breaking any scripts. A "major version" just means we have an interface-breaking change.